### PR TITLE
Traits refactoring to more specific names instead of ElementTrait and InnerElementTrait

### PR DESCRIPTION
**Pull request by bboro**
#614 [6h]

### CODE REVIEW

**Comment by bboro** [comment]
I think I prefer the latter, if we have an "InfoCard" paragraph type, and "InfoCardTrait", it makes sense that stuff for this paragraph type belongs there, and that's where we should start looking for things. What do you think?

**Comment by bboro** [comment]
3 / 6h will continue Monday. Think I'm heading in the right direction...

**Comment by amitaibu** [comment]
> should break it down further into "InfoCardTrait", "PeopleTrait" etc?

Yes, I agree we should break it down into smaller traits that are less abstract.

File: web/modules/custom/server_general/src/ElementLayoutTrait.php [hunk 1]
```diff
@@ -17,6 +19,7 @@
  */
 trait ElementLayoutTrait {
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementLayoutTrait.php:20]: I think this file is still abstracted. Maybe break it down as well?

File: web/modules/custom/server_general/src/InnerElementLayoutTrait.php [hunk 2]
```diff
@@ -8,15 +8,16 @@
 use Drupal\Core\Url;
 
 /**
- * Helper methods for rendering different Card layouts.
+ * Helper methods for rendering different "inner element" layouts such as cards.
  *
- * A card layout can be for example a card with an image, or a card with
- * centered items. This trait should only be used by `InnerElementTrait`.
+ * An inner element can be for example a card with an image, or a search result
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/InnerElementLayoutTrait.php:13]: 👍

**Review by amitaibu — APPROVED** [review]
Looks great, just a couple of comments, and MWG

Bumnp 1h to address

**Comment by bboro** [comment]
Summary of changes:

- Break down ElementTrait and InnerElementTrait into more specific traits under `Drupal\server_general\ElementTrait\` namespace such as `CtaTrait` and `CarouselTrait`.
- Renamed `buildInnerElement...` to `buildElement...` to keep it simpler
- Moved some more "layout-y" methods from  `ElementTrait` into `ElementLayoutTrait`
- Renamed `server_theme_inner_element__...` themes to `server_theme_element__...` themes to keep it simpler.

We still have some separation of "Element" and "InnerElement" though. The idea of "InnerElement" still exists in `InnerElementLayoutTrait` and contains some layout methods for Cards etc. I decided to keep this structuring separation for now, but we can also perhaps move this to `ElementLayoutTrait` later if needed. For now I updated the comment.

TB: 5 / 6h

**Comment by amitaibu** [comment]
> I decided to keep this structuring separation for now, but we can also perhaps move this to ElementLayoutTrait later if needed. For now I updated the comment.

Yeah, I think the fact you said it's a "private" trait is a good balance 👍

File: web/modules/custom/server_general/src/ElementLayoutTrait.php (hunk 1 above)
Comment (bboro) [reply web/modules/custom/server_general/src/ElementLayoutTrait.php:20]: I left it as it contains just "Layout" stuff now, such as TitleBodyAndItems etc. Where would you move these?

Comment (bboro) [reply web/modules/custom/server_general/src/ElementLayoutTrait.php:20]: I'm merging as is for now, we can break this down further later if you want to.

**Pull request merged**